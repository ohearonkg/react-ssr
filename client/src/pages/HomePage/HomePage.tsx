import * as React from "react";
import { Helmet } from "react-helmet";

const HomePage = () => {
  const buildHead = () => {
    return (
      <Helmet>
        <title> React SRR - Home </title>
        <meta name="og:title" content="React SSR - Home" />
      </Helmet>
    );
  };
  return (
    <div>
      {buildHead()}
      I'm The Home Component
    </div>
  );
};

export default {
  component: HomePage
};
