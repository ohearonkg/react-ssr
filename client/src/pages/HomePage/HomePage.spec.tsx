import { shallow, ShallowWrapper } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import HomePage from "./HomePage";

describe("Home Page", () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => (wrapper = shallow(<HomePage />)));

  /**
   * Snapshot
   */
  it("Should match its snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
