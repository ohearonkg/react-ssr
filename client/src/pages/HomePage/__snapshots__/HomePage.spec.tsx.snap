// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Should match its snapshot 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
       React SRR - Home 
    </title>
    <meta
      content="React SSR - Home"
      name="og:title"
    />
  </HelmetWrapper>
  I'm The Home Component
</div>
`;
